float32 right_knee
float32 right_hip
float32 left_knee
float32 left_hip
